src/toc_obj.rs

```diff
@@ -0,0 +1,42 @@
+use crate::traits::IntoString;
+use napi::bindgen_prelude::*;
+use visdom::Vis;
+
+#[napi(object)]
+#[derive(Debug)]
+pub struct TocObj {
+  pub text: String,
+  pub id: String,
+  pub level: u8,
+}
+
+#[napi]
+pub fn toc_obj(
+  #[napi(ts_arg_type = "Buffer | string")] input: Either3<Buffer, String, Unknown>,
+) -> Result<Vec<TocObj>> {
+  let input = input.to_string()?;
+  let root = Vis::load(&input).unwrap();
+  let lis = root.find(r#"[id^="title"]"#);
+
+  let mut result = vec![];
+  for element in lis {
```

Comment:
Any reason for not using `into_iter().map().collect()` here?

Reply:
Just a sample 😁